src/main/java/com/fasterxml/jackson/core/util/ObjectPool.java

```diff
@@ -0,0 +1,126 @@
+package com.fasterxml.jackson.core.util;
+
+import java.util.Deque;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public interface ObjectPool<T> extends AutoCloseable {
```

Comment:
Can we add javadoc? - `@since 2.16` and specified to be for internal use only

Reply:
Done.